corehq/apps/app_manager/suite_xml/sections/details.py

```diff
@@ -195,6 +195,10 @@ def build_detail(self, module, detail_type, detail, detail_column_infos, tabs=No
             if detail.lookup_enabled and detail.lookup_action:
                 d.lookup = self._get_lookup_element(detail, module)
 
+            # Add no items text
+            if detail_type.endswith('short'):
```

Comment:
I was following the pattern that I saw with the Case Claim flag where label for searching/again where the node always exists in the suite file. But the app_string toggles from custom/default text depending on if the flag is on.
`case_search.m0=Search All Cases`
`case_search.m0.again=Search Again`

Similarly in this case, the [app_string](https://github.com/dimagi/commcare-hq/blob/7b021ba211f698c4a2875f90ace58b4a088bd4cd/corehq/apps/app_manager/app_strings.py#L128-L136) the node always exists in the suite file but the app_string toggles to the default value (List is empty.) if the flag is off.

An alternative I can think of is passing the value directly from formplayer if the flag is off but that doesn't seem like a good place to define it.

Reply:
Ah, I get it now. So the functionality seems good, but there's a minor technical complexity tradeoff here: if you were to check for the flag here, that would make suite files a little bit smaller and let you avoid updating tests in this PR.

I was hoping it would also mean that the default text would only need to exist in the formplayer repo, not the HQ repo, but I see that it's useful for `Detail.no_items_text` to use the default string so the app builder knows what the default text is.

I'd still lean towards checking the flag here, because of the test file changes (smaller PRs are better), but reasonable people can disagree, defer to you on what to do.

Reply:
Agreed that smaller PRs are better - in the future I think it would be better if I had put test clean up changes like this in a separate PR to not muddy this PR for actual feature changes.

In this case, I will keep it as is. I think it will be easier to maintain by having the default defined in one location (in `Detail.no_items_text `) as opposed to defining it in two locations.

Reply:
Cool. FWIW, as a reviewer I think it's fine to put test & feature changes in the same PR. It's helpful to separate them by commit (as you have done), but multiple PRs that depend on each other are their own kind of complexity (in deploying/reverting).